rewrite-java/src/main/java/org/openrewrite/java/style/Autodetect.java

```diff
@@ -288,9 +288,15 @@ public TabsAndIndentsStyle getTabsAndIndentsStyle() {
                 tabSize = getClosestEven(x);
             }
 
-            if (tabSize < 2) {
-                tabSize = 2;
+            // Calculate tabSize based on the frequency, pick up the biggest frequency group.
+            // Using frequency are less susceptible to outliers than means.
+            int moreFrequentTabSize = getBiggestGroupOfTabSize(deltaSpaceIndentFrequencies);
+
+            // If two statistic results fight with each other, it means there are mess indentations in the code, then we use the default.
+            if (moreFrequentTabSize > 0 && tabSize != moreFrequentTabSize) {
+                tabSize = 4;
             }
```

Comment:
So is this doing two different calculations of tab size, checking that they agree, and if they do not defaulting to 4?
Seems like we should be able to pick a single method to determine tab size that is good enough

Reply:
Yes, it should be OK to use the max frequency group to determine, let me verify if it works, going to remove the previous methods.

Reply:
Yeah, removed the previous method of tabSize calculating, and kept the new method only.